app/resources/v1/geoip.js

```diff
@@ -0,0 +1,101 @@
+const config = require('config')
+const request = require('request')
+const logger = require('../../../lib/logger.js')()
+const redis = require('redis')
+const url = require('url')
+
+const IP_GEOLOCATION_TIMEOUT = 500
+
+exports.get = function (req, res) {
+  if (req.headers.host !== config.app_host_port) {
+    res.status(403).json({ status: 403, error: 'Not allowed to access API' })
+    return
+  }
+
+  const requestGeolocation = function (isRedisConnected = true) {
+    console.log('requesting geolocation')
+    const url = `${config.geoip.protocol}${config.geoip.host}?access_key=${config.geoip.api_key}`
+
+    request(url, { timeout: IP_GEOLOCATION_TIMEOUT }, function (error, response, body) {
+      if (error) {
+        logger.error(error)
+        // Error codes 101 and 104 are provided by ipstack to specify invalid/missing api key
+        // and maximum monthly API requests reached, respectively. Since ipstack already has
+        // error messages included with these errors, just send the message to the front-end.
+        if (error.code === 101 || error.code === 104) {
+          res.status(error.code).json({ status: error.code, error: error.info })
+        } else if (error.code === 'ETIMEDOUT') {
+          // If request takes longer than declared IP_GEOLOCATION_TIMEOUT, return status code 408
+          res.status(408).json({ status: 408, error: 'Request timed out' })
```

Comment:
I think we can abstract away some of the IP stack errors so that the front end does not need to know about things like missing API keys or usage limit being reached, because nothing on the front end can do anything about it. Meanwhile if we bucket everything under, say, a `503 Service Unavailable` that should be enough for the client side to handle.

As long as we've logged errors server side, we can track down ipstack errors without exposing the client to them.